railties/lib/rails/app_updater.rb

```diff
@@ -0,0 +1,47 @@
+require "rails/generators"
+require "rails/generators/rails/app/app_generator"
+
+module Rails
+  class AppUpdater # :nodoc:
+    class << self
+      def invoke_from_app_generator(method)
+        app_generator.send(method)
+      end
+
+      def app_generator
+        @app_generator ||= begin
+          gen = Rails::Generators::AppGenerator.new ["rails"], generator_options, destination_root: Rails.root
+          File.exist?(Rails.root.join("config", "application.rb")) ? gen.send(:app_const) : gen.send(:valid_const?)
+          gen
+        end
+      end
+
+      private
+        def generator_options
+          options = { api: !!Rails.application.config.api_only, update: true }
+
+          unless defined?(ActiveRecord)
```

Comment:
Can we make the assumption that if `ActiveRecord` is defined we are loading Active Record? I don't think this is true since a gem can define this constant. Maybe we should check for the railties?